src/smolagents/models.py

```diff
@@ -680,27 +681,34 @@ def __init__(
         if device_map is None:
             device_map = "cuda" if torch.cuda.is_available() else "cpu"
         logger.info(f"Using device: {device_map}")
-        self._is_vlm = False
-        try:
-            self.model = AutoModelForCausalLM.from_pretrained(
-                model_id,
-                device_map=device_map,
-                torch_dtype=torch_dtype,
-                trust_remote_code=trust_remote_code,
+
+        api = HfApi()
+        pipeline_tag = api.model_info(model_id).pipeline_tag
```

Comment:
We usually use the functions from the root of the package, so we use a single API instance:
```python
huggingface_hub.model_info
```